api/envoy/config/core/v3/base.proto

```diff
@@ -324,9 +324,15 @@ message HeaderValue {
   // The same :ref:`format specifier <config_access_log_format>` as used for
   // :ref:`HTTP access logging <config_access_log>` applies here, however
   // unknown header values are replaced with the empty string instead of ``-``.
-  string value = 2 [
-    (validate.rules).string = {max_bytes: 16384 well_known_regex: HTTP_HEADER_VALUE strict: false}
-  ];
+  oneof data {
```

Comment:
The HeaderValue message is used in other protobufs that are marked `stable`. Changing to oneof is a breaking change and we can not do it this way.

I think you need to drop oneof and use annotation: [(udpa.annotations.field_migrate).oneof_promotion = "value"];

Reply:
+1.  Besides, I think you also then should add config validation to enforce only one field is set.   

Basically:
1)  Add `[(udpa.annotations.field_migrate).oneof_promotion = "YOUR_NEW_NAME_FOR_ONEOF"` to the existing `value` field. Also, add the same annotation to the `value_bytes` field.
2) Add the comments saying only one of two is specified 
3) Add config validation to enforce this, throw exception otherwise.

BTW, [Here](https://github.com/envoyproxy/envoy/blob/main/api/STYLE.md#field-level) explains why in details.

Reply:
I see. More background about this: https://github.com/aip-dev/google.aip.dev/issues/393

Reply:
Thanks for the detail comments!
There is also an example here:  https://github.com/envoyproxy/envoy/blob/35514452d10ac56f5afca0c177149f0cdd030aa7/api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#L783.